scipy/spatial/tests/test_kdtree.py

```diff
@@ -1139,6 +1143,80 @@ def test_ckdtree_memuse():
     # ideally zero leaks, but errors might accidentally happen
     # outside cKDTree
     assert_(num_leaks < 10)
+
+def test_ckdtree_weights():
+
+    data = np.linspace(0, 1, 4).reshape(-1, 1)
+    tree1 = cKDTree(data, leafsize=1)
+    weights = np.ones(len(data), dtype='f4')
+
+    nw = tree1._build_weights(weights)
+    assert_array_equal(nw, [4, 2, 1, 1, 2, 1, 1])
+
+    try:
+        tree1._build_weights(weights[:-1])
+        raise AssertionError("Exception is not raised for short weights list")
+    except ValueError:
+        pass
```

Comment:
this block can be replaced by ``assert_raises(ValueError, tree1._build_weights, weights[:-1])``